src/org/openbravo/erpCommon/ad_actionButton/Posted.java

```diff
@@ -252,23 +258,24 @@ private void printPage(HttpServletResponse response, VariablesSecureApp vars, St
     }
 
     ActionButtonDefaultData[] data = null;
-    String strHelp = "", strDescription = "";
-    if (vars.getLanguage().equals("en_US")) {
+    String strHelp = "";
+    String strDescription = "";
+    if (StringUtils.equals("en_US", vars.getLanguage())) {
       data = ActionButtonDefaultData.select(this, strProcessId);
     } else {
       data = ActionButtonDefaultData.selectLanguage(this, vars.getLanguage(), strProcessId);
     }
 
-    if (data != null && data.length != 0) {
+    if (data.length != 0) {
       strDescription = data[0].description;
       strHelp = data[0].help;
     }
-    String[] discard = {"", ""};
-    if (strHelp.equals("")) {
-      discard[0] = new String("helpDiscard");
+    String[] discard = { "", "" };
```

Comment:
You can use StringUtils.EMPTY instead of ""

Reply:
Wouldn't make the code line unnecessarily longer and less readable?

Reply:
It's the same if you was used StringUtils.equals

Reply:
It's not the same to utilize the util for the equals (convention, plus null-check) than change two blank string parameters from a patch provided for a hotfix with a constant of StringUtils. 

String[] discard = { "", "" };
String[] discard = { StringUtils.BLANK, StringUtils.BLANK };

In this case I see it as unnecessarily long.

And in the other case, even more:

          HashMap<String, Integer> hm = ResetAccounting.delete(data[0].client, data[0].org, strTableId, strKey, "", "");

          HashMap<String, Integer> hm = ResetAccounting.delete(data[0].client, data[0].org, strTableId, strKey, StringUtils.BLANK, StringUtils.BLANK);

Reply:
And will make the code inconsistent accros the whole file. There are several blank strings in the entire file, changing them would be outside the scope of the hotfix.
 And when changing them, more issues will arise, also outside the scope.

A separate task should be done to refactor the whole class.

![image](https://github.com/etendosoftware/etendo_core/assets/84640069/f490b2ba-5b83-436e-be0d-56ae74798d48)